igcommit/commit_list_checks.py

```diff
@@ -253,3 +256,94 @@ def check_contributor(self, contributor, commit):
+        config_exists = new._prepare_configs(obj)
+        if config_exists:
+            return new
+
+    def _prepare_configs(self, commitlist):
+        """Update used configuration files, return true if any exist
+
+        This is copied from file_checks.py and slightly modified.
+        Please check the original file for explanation and
+        shortcomings.
+        """
+        config_exists = False
+        for config_file in self.config_files:
+            prev_commit = config_file.commit
+            # We will check if the config file exist on the latest commit
+            config_file.commit = commitlist[-1]
+
+            if config_file.exists():
+                config_exists = True
+
+                # If the file is not changed on this commit, we can skip
+                # downloading.
+                if not prev_commit or config_file.changed():
+                    with open(config_file.path, 'wb') as fd:
+                        fd.write(config_file.get_content())
+            elif exists(config_file.path):
+                # Not found on the latest commit, but it exists on the
+                # workspace.  Remove it.
+                remove(config_file.path)
+
+        return config_exists
+
+    def get_allowed_regexes(self):
+        for config_file in self.config_files:
+            with open(config_file.path) as f:
+                allowed_regexes = f.read().splitlines()
+        valid_list = []
+        failed_list = []
+        for regexline in allowed_regexes:
+            try:
+                if not regexline.strip():
+                    continue
+                re.compile(regexline)
+                valid_list.append(regexline)
+            except re.error:
+                failed_list.append(regexline)
+        return valid_list, failed_list
+
+    def check_branch_name(self, regexp_list, branch_name):
+        """Fails if the branch name does not match any regexp
+
+        It's enough to fit only one of the allowed patterns
+        """
+        trimmed_branch_name = re.sub('refs/heads/', '', branch_name)
+        match = False
+        for pattern in regexp_list:
+            if re.match(pattern, trimmed_branch_name):
+                match = True
+        return match
+
+    def get_problems(self):
+        branch_name = self.commit_list.branch_name
+        valid_regexes, broken_regexes = self.get_allowed_regexes()
+        if broken_regexes:
+            yield (
+                Severity.WARNING,
+                'following string is not a valid pattern '
+                'for branch name comparison:\n{}'
+                .format("\n".join(broken_regexes))
+            )
+        if not self.check_branch_name(valid_regexes, branch_name):
+            yield (
+                Severity.ERROR,
+                'branch name does not match any of the '
+                'alowed regexp patterns:\n{}'
+                .format("\n".join(valid_regexes))
+            )
```

Comment:
How about:
```
yield (
    Severity.ERROR,
    "{branch_name} is not an allowed branch name.\n"
    "Branch names must match any of the follow regular expressions:\n"
    "{valid_regexes.join('\n-')}"
)
```

Reply:
I refrained from f-strings like other functions, looks like we still support Python3.4. But I'll do the wording better :+1: